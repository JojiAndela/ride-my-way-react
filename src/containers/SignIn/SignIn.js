import React, { Component } from 'react';
import { connect } from 'react-redux';

import Form from 'Components/Forms/Form';
import Footer from 'Components/Forms/Footer';
import NavBar from 'Components/NavBar/Navbar';
import { loginUser } from 'Store/actions/auth'
import { updateObject, checkValidity } from '../../utility';

class SignIn extends Component {
  state={
    signInForm: {
      email: {
          inputType: 'input',
          elementConfig: {
              type: 'text',
              placeholder: 'Your E-Mail or Username'
          },
          value: '',
          validation: {
              required: true,
              isEmail: true
          },
          valid: false,
          touched: false
      },
      password: {
        inputType: 'input',
        elementConfig: {
            type: 'password',
            placeholder: 'Your Password'
        },
        value: '',
        validation: {
            required: true
        },
        valid: false,
        touched: false
    }
  }
  }
  onBtnClick = (e) => {
    e.preventDefault();
    let userData
    if (this.state.signInForm.email.value.includes("@")) {
      userData = {
        email: this.state.signInForm.email.value,
        password: this.state.signInForm.password.value
      };
    } else {
      userData = {
        username: this.state.signInForm.email.value,
        password: this.state.signInForm.password.value
      };
    }
      this.props.signinAction(userData, this.props.history);
  }
  inputChangedHandler = (event, inputIdentifier) => {

          const updatedFormElement = updateObject(this.state.signInForm[inputIdentifier], {
              value: event.target.value,
              valid: checkValidity(event.target.value, this.state.signInForm[inputIdentifier].validation),
          });
          const updatedOrderForm = updateObject(this.state.signInForm, {
              [inputIdentifier]: updatedFormElement
          });

          let formIsValid = true;
          for (let inputIdentifier in updatedOrderForm) {
              formIsValid = updatedOrderForm[inputIdentifier].valid && formIsValid;
          }
          this.setState({signInForm: updatedOrderForm, formIsValid: formIsValid});
      }
  render() {
    return (
      <div>
      <NavBar />
          <h1 style={{textAlign: 'center', color: '#bbb'}}>Sign In</h1>
        <Form form={this.state.signInForm} btnName="Sign In" btnClick={this.onBtnClick} onChanged={this.inputChangedHandler}/>
          <Footer type="signin" />
      </div>
    );
}
}

const mapDispatchToProps = dispatch => ({
  signinAction: (userData, history) => dispatch(loginUser(userData, history)),
});

export default connect(null, mapDispatchToProps)(SignIn);
