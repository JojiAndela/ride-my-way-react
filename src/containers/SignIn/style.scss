.signin{
  width: 50%;
  margin: auto;
}
@media screen and (max-width: 1090px){
  .signin{
    width:100%;
  }
}
