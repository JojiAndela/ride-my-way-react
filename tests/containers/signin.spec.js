import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

// Component to be tested
import SignIn from '../../src/containers/SignIn/SignIn';

const initialState ={
  rides: {
    loading: false,
    rides: [],
    singleRide: null,
    requests: [],
    myRequest: {}
  }
}
const state = {signUpForm:{
  name: {
      inputType: 'input',
      elementConfig:{
          type: 'text',
          placeholder: 'Your Username',
          minLength: "3",
          required: true,
      },
      value: '',
      validation: {
          required: true
      },
      valid: false,
      touched: false,
  },
}}

const onBtnClick = jest.fn();
const inputChangedHandler = jest.fn();
const middleware = [thunk];
const mockStore = configureStore(middleware);
const signupAction = jest.fn();
const setErrorAction = jest.fn();

let wrapper;
let store;
describe('SignIn', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <SignIn onBtnClick={onBtnClick}
            inputChangedHandler={inputChangedHandler}
            signupAction={signupAction}
            setErrorAction={setErrorAction}
          />
        </BrowserRouter>
      </Provider>
    );
  });
  it('should render the sign in page correctly', () => {
    wrapper.setState(state);
    expect(wrapper).toMatchSnapshot();
  });
});
