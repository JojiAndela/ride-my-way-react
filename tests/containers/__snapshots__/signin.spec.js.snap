// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should render the sign in page correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(SignIn)
        inputChangedHandler={[MockFunction]}
        onBtnClick={[MockFunction]}
        setErrorAction={[MockFunction]}
        signupAction={[MockFunction]}
      >
        <SignIn
          inputChangedHandler={[MockFunction]}
          onBtnClick={[MockFunction]}
          setErrorAction={[MockFunction]}
          signinAction={[Function]}
          signupAction={[MockFunction]}
        >
          <div>
            <Navbar>
              <div
                className="Navbar"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="Logo"
                  to="/"
                >
                  <Route
                    path="\\\\/"
                  >
                    <Link
                      aria-current="page"
                      className="Logo active"
                      replace={false}
                      style={Object {}}
                      to="/"
                    >
                      <a
                        aria-current="page"
                        className="Logo active"
                        href="/"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        Ride My Way
                      </a>
                    </Link>
                  </Route>
                </NavLink>
                <nav
                  className="NavItems"
                >
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/about"
                  >
                    <Route
                      path="\\\\/about"
                    >
                      <Link
                        aria-current={null}
                        replace={false}
                        to="/about"
                      >
                        <a
                          aria-current={null}
                          href="/about"
                          onClick={[Function]}
                        >
                          About
                        </a>
                      </Link>
                    </Route>
                  </NavLink>
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/signin"
                  >
                    <Route
                      path="\\\\/signin"
                    >
                      <Link
                        aria-current={null}
                        replace={false}
                        to="/signin"
                      >
                        <a
                          aria-current={null}
                          href="/signin"
                          onClick={[Function]}
                        >
                          Sign in
                        </a>
                      </Link>
                    </Route>
                  </NavLink>
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/dashboard"
                  >
                    <Route
                      path="\\\\/dashboard"
                    >
                      <Link
                        aria-current={null}
                        replace={false}
                        to="/dashboard"
                      >
                        <a
                          aria-current={null}
                          href="/dashboard"
                          onClick={[Function]}
                        >
                          Dashboard
                        </a>
                      </Link>
                    </Route>
                  </NavLink>
                </nav>
              </div>
            </Navbar>
            <h1
              style={
                Object {
                  "color": "#bbb",
                  "textAlign": "center",
                }
              }
            >
              Sign In
            </h1>
            <form
              btnClick={[Function]}
              btnName="Sign In"
              form={
                Object {
                  "email": Object {
                    "elementConfig": Object {
                      "placeholder": "Your E-Mail or Username",
                      "required": true,
                      "type": "text",
                    },
                    "inputType": "input",
                    "valid": false,
                    "validation": Object {
                      "isEmail": true,
                      "required": true,
                    },
                    "value": "",
                  },
                  "password": Object {
                    "elementConfig": Object {
                      "placeholder": "Your Password",
                      "required": true,
                      "type": "password",
                    },
                    "inputType": "input",
                    "valid": false,
                    "validation": Object {
                      "required": true,
                    },
                    "value": "",
                  },
                }
              }
              onChanged={[Function]}
            >
              <form
                className="Form"
                onSubmit={[Function]}
              >
                <input
                  changed={[Function]}
                  elementConfig={
                    Object {
                      "placeholder": "Your E-Mail or Username",
                      "required": true,
                      "type": "text",
                    }
                  }
                  inputType="input"
                  key="email"
                  value=""
                >
                  <input
                    className="input"
                    onChange={[Function]}
                    placeholder="Your E-Mail or Username"
                    required={true}
                    type="text"
                    value=""
                  />
                </input>
                <input
                  changed={[Function]}
                  elementConfig={
                    Object {
                      "placeholder": "Your Password",
                      "required": true,
                      "type": "password",
                    }
                  }
                  inputType="input"
                  key="password"
                  value=""
                >
                  <input
                    className="input"
                    onChange={[Function]}
                    placeholder="Your Password"
                    required={true}
                    type="password"
                    value=""
                  />
                </input>
                <button
                  btnType="Success"
                >
                  <button
                    className="Button Success"
                  >
                    Sign In
                  </button>
                </button>
              </form>
            </form>
            <footer
              type="signin"
            >
              <div>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="FooterLink"
                  to="/signup"
                >
                  <Route
                    path="\\\\/signup"
                  >
                    <Link
                      aria-current={null}
                      className="FooterLink"
                      replace={false}
                      to="/signup"
                    >
                      <a
                        aria-current={null}
                        className="FooterLink"
                        href="/signup"
                        onClick={[Function]}
                      >
                        New here? Sign Up!
                      </a>
                    </Link>
                  </Route>
                </NavLink>
              </div>
            </footer>
          </div>
        </SignIn>
      </Connect(SignIn)>
    </Router>
  </BrowserRouter>
</Provider>
`;
